PWGCF/EbyEFluctuations/Tasks/Diff_pT_fluct_PID.cxx

```diff
@@ -0,0 +1,758 @@
+
+        fD_proton = fB1_proton * fB2_proton;
+        histos.fill(HIST("fD_proton"), Cent_FT0M, 0.5, fD_proton);
+        Subsample[SampleIndex][19]->Fill(Cent_FT0M, 0.5, fD_proton);
+      }
+
+      for (int i = 0; i < 14; i++) {
+        // A_hadrone Calculation
+        if (Nch_eta_less_0 != 0) {
+          fA_hadron[i] = fPt_profile->GetBinContent(i + 1) / Nch_eta_less_0;
+          histos.fill(HIST("fA_hadron"), Cent_FT0M, pT_bin[i], fA_hadron[i]);
+          Subsample[SampleIndex][0]->Fill(Cent_FT0M, pT_bin[i], fA_hadron[i]);
+        }
+
+        // A_pion Calculation
+        if (N_Pi_eta_less_0 != 0) {
+          fA_pion[i] = fPt_profile_pion->GetBinContent(i + 1) / N_Pi_eta_less_0;
+          histos.fill(HIST("fA_pion"), Cent_FT0M, pT_bin[i], fA_pion[i]);
+          Subsample[SampleIndex][1]->Fill(Cent_FT0M, pT_bin[i], fA_pion[i]);
+        }
+
+        // A_kaon Calculation
+        if (N_Ka_eta_less_0 != 0) {
+          fA_kaon[i] = fPt_profile_kaon->GetBinContent(i + 1) / N_Ka_eta_less_0;
+          histos.fill(HIST("fA_kaon"), Cent_FT0M, pT_bin[i], fA_kaon[i]);
+          Subsample[SampleIndex][2]->Fill(Cent_FT0M, pT_bin[i], fA_kaon[i]);
+        }
+
+        // A_proton Calculation
+        if (N_Pr_eta_less_0 != 0) {
+          fA_proton[i] = fPt_profile_proton->GetBinContent(i + 1) / N_Pr_eta_less_0;
+          histos.fill(HIST("fA_proton"), Cent_FT0M, pT_bin[i], fA_proton[i]);
+          Subsample[SampleIndex][3]->Fill(Cent_FT0M, pT_bin[i], fA_proton[i]);
+        }
+
+        // C_hadron Calculation
+        if (Nch_eta_less_0 != 0 && N_sum_etaGreaterEtamin != 0) {
+          fC_hadron[i] = fA_hadron[i] * fB2_hadron;
+          histos.fill(HIST("fC_hadron"), Cent_FT0M, pT_bin[i], fC_hadron[i]);
+          Subsample[SampleIndex][12]->Fill(Cent_FT0M, pT_bin[i], fC_hadron[i]);
+        }
+
+        // C_pion Calculation
+        if (N_Pi_eta_less_0 != 0 && N_sum_etaGreaterEtamin != 0) {
+          fC_pion[i] = fA_pion[i] * fB2_pion;
+          histos.fill(HIST("fC_pion"), Cent_FT0M, pT_bin[i], fC_pion[i]);
+          Subsample[SampleIndex][13]->Fill(Cent_FT0M, pT_bin[i], fC_pion[i]);
+        }
+
+        // A_kaon Calculation
+        if (N_Ka_eta_less_0 != 0 && N_sum_etaGreaterEtamin != 0) {
+          fC_kaon[i] = fA_kaon[i] * fB2_kaon;
+          histos.fill(HIST("fC_kaon"), Cent_FT0M, pT_bin[i], fC_kaon[i]);
+          Subsample[SampleIndex][14]->Fill(Cent_FT0M, pT_bin[i], fC_kaon[i]);
+        }
+
+        // A_proton Calculation
+        if (N_Pr_eta_less_0 != 0 && N_sum_etaGreaterEtamin != 0) {
+          fC_proton[i] = fA_proton[i] * fB2_proton;
+          histos.fill(HIST("fC_proton"), Cent_FT0M, pT_bin[i], fC_proton[i]);
+          Subsample[SampleIndex][15]->Fill(Cent_FT0M, pT_bin[i], fC_proton[i]);
+        }
+      }
+
+      fPt_profile->Delete();
+      fPt_profile_pion->Delete();
+      fPt_profile_kaon->Delete();
+      fPt_profile_proton->Delete();
```

Comment:
Wouldn't be more efficient to keep the profiles created and just clear them on each process call?